app/controllers/Authentication.scala

```diff
@@ -75,6 +76,7 @@ class Authentication @Inject() (
           profile <- p.retrieveProfile(authInfo.asInstanceOf[p.A])
           customProfile = profile match {
             case p: CustomSocialProfile => p
+            case p: CommonSocialProfile => CustomSocialProfile.fromCommonSocialProfile(p)
```

Comment:
Twitter won't return gender, so no need to write a custom profile parser that will return custom profile. We just convert the common profile to a custom one (with gender set to None)